### ISSUE TITLE
[Question] what is reward scaling ?

### ISSUE THREAD

**Issue by pixar0407** [seolhokim/Mujoco-Pytorch#1]
Hello, seolhokim.

thanks for the quality code.
I have a question. what is args.reward_scaling ? And why did you choose 0.1 for the reward_scaling?
I found the possible hint from this paper (https://openreview.net/pdf?id=r1etN1rtPB). I cannot pinpoint the exact relationship with args.reward_scaling? On the appendix A.2, reward is divided by RS. But I cannot relate RS with args.reward_scaling?
So, please tell me what is args.reward_scaling and why we use it, and why did you choose 0.1 for it?

좋은 코드 감사합니다.
질문이 있어서 이렇게 글을 남깁니다. args.reward_scaling은 무엇이며, 왜 0.1로 설정하였나요?
검색을 하다가 이 논문([link](https://openreview.net/pdf?id=r1etN1rtPB))에서 힌트를 찾은것 같은데요. 사실 논문과 seolhokim님이 짠 코드와 연결을 못 시키겠습니다.
Appendix A.2에 reward를 RS로 나누는거 같은데, 이게 0.1(args.reward_scaling)과 관련있는거 같진 않습니다.
혹시 괜찮으시면, reward_scaling의 정체와 우리가 왜 이걸 사용해야 하는지, 그리고 왜 0.1로 설정했는지 가르쳐 주실 수 있으신가요?

**Comment by seolhokim** [comment]
@pixar0407 
Hi, Thank you for your interest. When training agents, you should pay attention to network training(optimization) problems. You need to keep in mind excessive cumulative reward can diverge the network.

The paper you found describes similar techniques for stable training. Proper scaling can help your agent learning.

**Comment by pixar0407** [comment]
@seolhokim, Thanks for the reply. 
So, Do you mean that it is hyperparameter? 
But how did you choose it as 0.1? I cannot find it on baseline3[link](https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/ppo/ppo.py).

Is it empirical? And is it changed depend on environment(e.g. hooper, walker2d etc) or network?

그럼 reward_scaling은 그저 hyperparameter란 뜻인가요? 
하지만 어떤 과정을 거쳐서 0.1이라는 숫자를 얻게 되셨나요? 공식 code인 baseline3도 나와 있지 않습니다.
여러번 실험을 거쳐서 0.1이 최선임을 알게 되셨나요?  그리고 reward_scaling은 게임의 종류(hooper, walker2d 등)에 따라 달라지나요? 아니면 network에 따라 달라지나요?

**Comment by pixar0407** [comment]
@seolhokim 

And, Can you tell me the difference between reward_scaling and gamma?
I know that gamma also works for smoothing excessive cumulative reward.

아 그리고, reward_scaling과 감마와의 역할 차이 알려주실 수 있나요?
excessive cumulative reward를 경감시키는 방안은 감마가 해주는 것 아닌가요?
reward_scaling은 reward 각각의 scaling 자체를 1/10 해주는 것인데, 감마는 미래의 reward의 영향을 감소하는 것 아닌가요?

**Comment by seolhokim** [comment]
@pixar0407 
You understand properly. It usually depends on environment(ex : environment episode length). You can choose the appropriate scaling parameter by looking at the value estimation in training process. You can find some paper that used reward scaling parameter(but more delicate like you found). Finding the reward scaling parameter value just relied on my intuition.

Suppose the environments that agent always get reward 10 and 1. If gamma is 0.9, then agent's value estimations are 100 and 10. In my point of view, 100 is too high. Not only the future effect of reward, but also the current effect decreases.

**Comment by pixar0407** [comment]
@seolhokim thank you for the quick answers.

Now i finally come to understand the reward_scaling. 
But i cannot understand the what you said.
> Suppose the environments that agent always get reward 10 and 1. If gamma is 0.9, then agent's value estimations are 100 and 10.

Can you show how did you calculate 100 out of 0.9 and 10? Sorry for the bothering. It seem somewhat 10/(1-0.9)=100. But i don't understand the equation's meaning.

답변 감사합니다. 이제 reward_scaling은 이해되었습니다. 
그런데, 답변 중 아래의 문장이 이해하기 힘듭니다.
> Suppose the environments that agent always get reward 10 and 1. If gamma is 0.9, then agent's value estimations are 100 and 10.

어떻게 value estimation 100이 0.9와 10으로 부터 나왔는지 공식으로 설명가능하신가요? 귀찮게 해드려 죄송합니다. 
10/(1-0.9)=100으로 계산했을 것 같은데, 이 공식의 의미를 모르겠습니다.

**Comment by seolhokim** [comment]
@pixar0407 
Suppose your agent estimates value in state s1. Then exact value estimation about s1 is 10 + 10 x 0.9 + 10 x 0.9^2 + ... = 100. Sorry for making you misunderstand.

**Comment by pixar0407** [comment]
@seolhokim 

Ah, now it makes sense.
Thank you for answering all the continuing questions. 
Hope you have good weekend night.
